src/Endpoint/AbstractEndpoint.php

```diff
@@ -63,6 +123,21 @@ protected function execute($method, $verb, array $data)
     }
 
     /**
+     * Returns the response for a synchronous request
+     *
+     * @param string $method HTTP method to use for the request
+     * @param string $verb   ThisData verb to be used on this endpoint
+     * @param array  $data   Request body data containing event metadata
+     */
+    protected function synchronousExecute($method, $verb, array $data = [])
```

Comment:
This completely bypasses the configured handler. E.g. as a developer, if I want to make this request asynchronously, which the documentation would lead you to believe it possible, you're out of luck.

I think there needs to be more thought put into how this is implemented. This will get the job done for now until I can think of a better solution however. Just wanted to raise it.

The [synchronous client's send method just calls async send with a wait](https://github.com/guzzle/guzzle/blob/master/src/Client.php#L101-L105). Perhaps we could expose this in a clean way so that regardless of the handler, we can expect simple access to the response.